bug: TestTutorial01/controller flakes on controller.txtar:28 — standalone controller PID exit timeout

### Before you continue

- [x] I searched existing issues and did not find a duplicate (closest: #562, distinct — see Additional context).
- [x] I read the relevant docs and contributor guidance.

### Gas City version

main@b1d5ed1f (observed on PR #548 head)

### Environment

GitHub Actions CI — Linux runner. Test path `cmd/gc` using `GC_SESSION=fake` (fake session provider).

### Reproduction

Observed on a CI run against PR #548 (branch `fix/532-flaky-stop-concurrent-sessions`). It is a flake — has not been bisected to a deterministic repro, but the timing pattern is distinctive and may help.

To attempt reproduction locally:

```bash
go test ./cmd/gc -run 'TestTutorial01/controller' -count=50 -timeout 600s -v
```

The txtar script under test is `cmd/gc/testdata/controller.txtar`. It starts a standalone controller with `GC_SESSION=fake`, stops it, waits, then starts a second controller, verifies a double-start is rejected, and finally cleans up the first with a second `gc stop` (line 28). The second `gc stop` is the one that flakes.

### Expected behavior

`gc stop $WORK/ctrl-city` on line 28 succeeds — the standalone controller exits promptly after being signaled to stop, `gc stop` returns exit 0, and the `wait ctrl2` on line 29 completes.

### Actual behavior

The controller prints `Controller stopping...` to stdout (so it received and began processing the stop request) but then does not exit in time, and `gc stop` gives up waiting for the PID:

```
gc stop: timed out waiting for standalone controller (PID 40995) to stop
```

The whole `TestTutorial01/controller` subtest took 20.55s, which is abnormal — healthy runs should be well under that. The earlier first-stop on line 12 also took 7.741s (also slow), suggesting lifecycle is already marginal before it tips over on the second stop.

### Logs, screenshots, or traces

```shell
--- FAIL: TestTutorial01 (0.28s)
    --- FAIL: TestTutorial01/controller (20.55s)
        testscript.go:584: WORK=/tmp/gc-testscript-797313398/script-controller
            # Controller mode tests with fake session provider. (0.000s)
            # gc start --controller runs the persistent loop and can be stopped. (1.004s)
            # gc stop connects to the controller socket and asks it to shut down. (7.741s)   <-- slow
            # Wait for the controller to finish. (0.000s)
            # Double-start: start one controller, try to start another. (1.029s)
            # Clean up the first controller. (10.779s)                                         <-- timeout
            > exec gc stop \$WORK/ctrl-city
            [stdout]
            Controller stopping...
            [stderr]
            gc stop: timed out waiting for standalone controller (PID 40995) to stop
            [exit status 1]
            FAIL: testdata/controller.txtar:28: unexpected command failure
            [background] gc start --controller \$WORK/ctrl-city: exit status 0
            [stdout]
            Controller started.
            Controller stopped.
FAIL
coverage: 69.7% of statements
FAIL	github.com/gastownhall/gascity/cmd/gc	55.010s
```

Full failed CI job log: https://github.com/gastownhall/gascity/actions/runs/24239714507/job/70771015954

### Additional context

- Observed on PR #548 (https://github.com/gastownhall/gascity/pull/548), but **unrelated** to that PR's changes. PR #548 touches only `internal/runtime/acp/acp.go` (a single-line reorder in the real ACP session provider). `controller.txtar` uses `GC_SESSION=fake` (see lines 1–5 of the file), so it never executes the ACP provider code path — this failure cannot be a regression from that PR.
- Related but distinct from #562, which tracks:
  - `TestControllerShutdown` in `cmd/gc/controller_test.go:165` — "timed out waiting for controller socket to become available" (startup readiness race), and
  - `controller.txtar:13` — output drift (`City stopped.` vs expected `Controller stopping`).
  - This bug is instead at `controller.txtar:28` (cleanup stop) and the symptom is shutdown-side timeout waiting for the controller PID to exit *after* the stop signal was acknowledged. They may share a root cause (standalone controller lifecycle race) but the failure modes are distinguishable, so filing separately.
- Priority is medium — it's a test flake, not a runtime regression, but it will continue to cause spurious CI failures on unrelated PRs until addressed.

--- a/cmd/gc/city_runtime.go
+++ b/cmd/gc/city_runtime.go
@@ -412,6 +412,9 @@ func (cr *CityRuntime) tick(
 	if dirty.Swap(false) {
 		cr.reloadConfigTraced(ctx, lastProviderName, cityRoot, trace)
 	}
+	if ctx.Err() != nil {
+		return
+	}
 
 	// Session bead sync BEFORE reconciliation (one-tick state lag; see run()).
 	// Post-reconcile sync was intentionally removed: the daemon's next tick
@@ -439,6 +442,9 @@ func (cr *CityRuntime) tick(
 	if cr.sessionDrains != nil {
 		cr.beadReconcileTick(ctx, result, sessionBeads, trace)
 	}
+	if ctx.Err() != nil {
+		return
+	}
 
 	// Wisp GC: purge expired closed molecules.
 	if cr.wg != nil && cr.wg.shouldRun(time.Now()) {
@@ -450,9 +456,16 @@ func (cr *CityRuntime) tick(
 		}
 	}
 
+	if ctx.Err() != nil {
+		return
+	}
+
 	// Order dispatch.
 	if cr.od != nil {
 		cr.od.dispatch(ctx, cityRoot, time.Now())
+	}
+	if ctx.Err() != nil {
+		return
 	}
 
 	if cr.svc != nil {
@@ -462,6 +475,9 @@ func (cr *CityRuntime) tick(
 	// Chat session auto-suspend: suspend detached idle sessions.
 	if idleTimeout := cr.cfg.ChatSessions.IdleTimeoutDuration(); idleTimeout > 0 {
 		autoSuspendChatSessions(cr.cityBeadStore(), cr.sp, idleTimeout, clock.Real{}, cr.stdout, cr.stderr)
+	}
+	if ctx.Err() != nil {
+		return
 	}
 
 	// Drain queued convergence requests (CLI commands) BEFORE tick so

--- a/cmd/gc/cmd_stop.go
+++ b/cmd/gc/cmd_stop.go
@@ -69,7 +69,7 @@ func cmdStop(args []string, stdout, stderr io.Writer) int {
 
 	// If a controller is running, ask it to shut down (it stops agents).
 	if tryStopController(cityPath, stdout) {
-		if err := waitForStandaloneControllerStop(cityPath, cfg.Daemon.ShutdownTimeoutDuration()+5*time.Second); err != nil {
+		if err := waitForStandaloneControllerStop(cityPath, cfg.Daemon.ShutdownTimeoutDuration()+15*time.Second); err != nil {
 			fmt.Fprintf(stderr, "gc stop: %v\n", err) //nolint:errcheck // best-effort stderr
 			return 1
 		}

--- a/cmd/gc/cmd_stop_test.go
+++ b/cmd/gc/cmd_stop_test.go
@@ -187,3 +187,103 @@ func TestCmdStopUsesTargetCitySessionProviderOutsideCityDir(t *testing.T) {
 		t.Fatalf("session provider provider = %q, want %q", gotProvider, "subprocess")
 	}
 }
+
+// TestCmdStopMarginExhaustion verifies that cmdStop tolerates slow controller
+// shutdowns without timing out. With a non-zero ShutdownTimeout and a provider
+// whose Stop blocks briefly (simulating CI scheduling delays or an in-flight
+// tick), the increased wait margin must absorb the overhead.
+//
+// Regression test for gastownhall/gascity#572.
+func TestCmdStopMarginExhaustion(t *testing.T) {
+	t.Setenv("GC_HOME", shortSocketTempDir(t, "gc-home-"))
+
+	dir := shortSocketTempDir(t, "gc-margin-")
+	if err := os.MkdirAll(filepath.Join(dir, ".gc"), 0o755); err != nil {
+		t.Fatal(err)
+	}
+
+	cfg := &config.City{
+		Workspace: config.Workspace{Name: "test-margin"},
+		Beads:     config.BeadsConfig{Provider: "file"},
+		Daemon:    config.DaemonConfig{ShutdownTimeout: "1s"},
+	}
+	data, err := cfg.Marshal()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := os.WriteFile(filepath.Join(dir, "city.toml"), data, 0o644); err != nil {
+		t.Fatal(err)
+	}
+
+	sp := newGatedStopProvider()
+	buildFn := func(_ *config.City, _ runtime.Provider, _ beads.Store) DesiredStateResult {
+		return DesiredStateResult{State: map[string]TemplateParams{}}
+	}
+
+	var controllerStdout, controllerStderr bytes.Buffer
+	done := make(chan struct{})
+	go func() {
+		runController(dir, filepath.Join(dir, "city.toml"), cfg, "", buildFn, nil, sp, nil, nil, nil, nil, events.Discard, nil, &controllerStdout, &controllerStderr)
+		close(done)
+	}()
+	t.Cleanup(func() {
+		running, _ := sp.ListRunning("")
+		for _, name := range running {
+			sp.release(name)
+		}
+		tryStopController(dir, &bytes.Buffer{})
+		select {
+		case <-done:
+		case <-time.After(5 * time.Second):
+		}
+	})
+
+	waitForController(t, dir, 5*time.Second)
+
+	const sess = "margin-session"
+	if err := sp.Start(context.Background(), sess, runtime.Config{}); err != nil {
+		t.Fatal(err)
+	}
+
+	go func() {
+		sp.waitForInterrupts(t, 1)
+		sp.releaseInterrupt(sess)
+	}()
+
+	var stdout, stderr bytes.Buffer
+	stopDone := make(chan int, 1)
+	go func() {
+		stopDone <- cmdStop([]string{dir}, &stdout, &stderr)
+	}()
+
+	stopped := sp.waitForStops(t, 1)
+	if len(stopped) != 1 || stopped[0] != sess {
+		t.Fatalf("stop targets = %v, want [%s]", stopped, sess)
+	}
+
+	time.AfterFunc(500*time.Millisecond, func() {
+		sp.release(sess)
+	})
+
+	select {
+	case code := <-stopDone:
+		if code != 0 {
+			t.Fatalf("cmdStop = %d, want 0; stdout=%q stderr=%q", code, stdout.String(), stderr.String())
+		}
+	case <-time.After(20 * time.Second):
+		t.Fatal("cmdStop did not finish within margin budget")
+	}
+
+	select {
+	case <-done:
+	case <-time.After(5 * time.Second):
+		t.Fatal("controller did not exit after cmdStop")
+	}
+
+	if !strings.Contains(stdout.String(), "Controller stopping...") {
+		t.Fatalf("stdout missing controller stop message: %q", stdout.String())
+	}
+	if !strings.Contains(stdout.String(), "City stopped.") {
+		t.Fatalf("stdout missing city stopped message: %q", stdout.String())
+	}
+}

--- a/cmd/gc/cmd_stop_test.go
+++ b/cmd/gc/cmd_stop_test.go
@@ -220,7 +220,8 @@ func TestCmdStopMarginExhaustion(t *testing.T) {
 		return DesiredStateResult{State: map[string]TemplateParams{}}
 	}
 
-	var controllerStdout, controllerStderr bytes.Buffer
+	var controllerStdout bytes.Buffer
+	var controllerStderr syncBuffer
 	done := make(chan struct{})
 	go func() {
 		runController(dir, filepath.Join(dir, "city.toml"), cfg, "", buildFn, nil, sp, nil, nil, nil, nil, events.Discard, nil, &controllerStdout, &controllerStderr)
@@ -238,7 +239,7 @@ func TestCmdStopMarginExhaustion(t *testing.T) {
 		}
 	})
 
-	waitForController(t, dir, 5*time.Second)
+	waitForController(t, dir, 5*time.Second, done, &controllerStderr)
 
 	const sess = "margin-session"
 	if err := sp.Start(context.Background(), sess, runtime.Config{}); err != nil {

--- a/test/docsync/docsync_test.go
+++ b/test/docsync/docsync_test.go
@@ -34,7 +34,7 @@ var docTreeDirs = []string{"contrib", "docs", "engdocs"}
 
 // docTreeIgnored lists directories that contain markdown but are not
 // documentation trees (e.g., embedded prompt templates, test fixtures).
-var docTreeIgnored = []string{"cmd", "examples", "internal", "scripts", "test"}
+var docTreeIgnored = []string{"cmd", "examples", "internal", "plans", "scripts", "test"}
 
 // knownBrokenLinks lists links to docs that do not exist yet. These are
 // excluded from TestLocalMarkdownLinks failures but still logged. Remove

